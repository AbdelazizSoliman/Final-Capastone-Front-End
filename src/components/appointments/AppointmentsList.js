import React from 'react';

const AppointmentsList = () => (
  <div>
    Appointments List
  </div>
);

export default AppointmentsList;
