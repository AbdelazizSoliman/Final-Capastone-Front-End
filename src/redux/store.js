import { configureStore } from '@reduxjs/toolkit';
import homeReducer from './reducers/homeSlice';

const Store = configureStore({
  reducer: {
    doctors: homeReducer,
  },
});

export default Store;
