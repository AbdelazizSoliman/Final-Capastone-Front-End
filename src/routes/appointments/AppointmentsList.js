import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import axios from 'axios';
import { fetchAppointments } from '../../redux/appointments/appointmentThunk';
import SideNav from '../../components/home/SideNav';

const AppointmentsList = () => {
  const dispatch = useDispatch();
  const [appointmentsArr, setAppointmentsArray] = useState([]);
  const [displayedAppointments, setDisplayedAppointments] = useState([]);
  const [currentIndex, setCurrentIndex] = useState(0);

  useEffect(() => {
    if (appointmentsArr.length === 0 || appointmentsArr.length === undefined) {
      dispatch(fetchAppointments());
    }
  }, [dispatch, appointmentsArr]);

  useEffect(() => {
    axios.get('http://localhost:3000/api/v1/appointments')
      .then((response) => {
        setAppointmentsArray(response.data);
        setDisplayedAppointments(response.data.slice(0, 3));
      })
      .catch((error) => {
        throw new Error(error.response?.data?.error || 'Failed to fetch appointments');
      });
  }, []);

  const shiftAppointments = (direction) => {
    const increment = direction === 'left' ? -1 : 1;
    const newIndex = (currentIndex + increment + appointmentsArr.length) % appointmentsArr.length;
    setCurrentIndex(newIndex);
    const endIndex = newIndex + 3 >= appointmentsArr.length ? appointmentsArr.length : newIndex + 3;
    setDisplayedAppointments(appointmentsArr.slice(newIndex, endIndex));
  };

  return (
    <div className="h-screen d-flex">
      <SideNav />

      <div className="p-4" style={{ flex: '1' }}>
        <h3 className="text-primary">Appointments List</h3>
        <div className="d-flex align-items-center" style={{ overflowY: 'hidden' }}>
          <button className="btn btn-link" type="button" onClick={() => shiftAppointments('left')}>
            <i className="fa fa-chevron-left" style={{ fontSize: '1.5em', color: '#25c804' }} />
          </button>
          <div className="horizontal-scroll" style={{ overflowX: 'auto', whiteSpace: 'nowrap' }}>
            <ul className="list-group d-flex flex-row">
              {displayedAppointments.map((appointment) => (
                <li key={appointment.id} className="list-group-item m-2 p-2" style={{ width: '300px' }}>
                  <h5 className="text-info">Appointment Details:</h5>
                  <p>
                    <strong className="text-warning">Date:</strong>
                    {' '}
                    {appointment.date_of_appointment}
                  </p>
                  <p>
                    <strong className="text-success">Time:</strong>
                    {' '}
                    {appointment.time_of_appointment}
                  </p>
                  <p>
                    <strong className="text-primary">City:</strong>
                    {' '}
                    {appointment.city}
                  </p>
                  <p>
                    <strong className="text-primary">Doctor:</strong>
                    {' '}
                    {appointment.doctor_name}
                  </p>
                  <p>
                    <strong className="text-primary">Patient:</strong>
                    {' '}
                    {appointment.patient_name}
                  </p>
                </li>
              ))}
            </ul>
          </div>
          <button className="btn btn-link" type="button" onClick={() => shiftAppointments('right')}>
            <i className="fa fa-chevron-right" style={{ fontSize: '2em', color: '#25c804' }} />
          </button>
        </div>
      </div>
    </div>
  );
};

export default AppointmentsList;
