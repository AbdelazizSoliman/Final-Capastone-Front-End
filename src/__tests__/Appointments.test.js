import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import AppointmentsList from '../routes/appointments/AppointmentsList';
import store from '../redux/store';

describe('Appointmetns component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <AppointmentsList />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
