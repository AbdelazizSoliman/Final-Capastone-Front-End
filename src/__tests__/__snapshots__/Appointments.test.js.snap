// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointmetns component should render correctly 1`] = `
<div>
  <div
    class="d-flex w-100 align-items-center justify-content-center bg-green"
  >
    <div
      class="d-flex flex-column "
    >
      <img
        alt="getDocLogo"
        class="img-fluid my-4"
        src="getdoc.png"
        style="max-width: 40%;"
      />
      <ul
        class="list-group text-center border-0 mt-1"
      >
        <li
          class="li-cont list-group-item border-0 pl-2 fw-bold d-flex"
        >
          <a
            class="link text-decoration-none align-self-start"
            href="/"
          >
            DOCTORS
          </a>
        </li>
        <li
          class="list-group-item custom-link border-0 pl-2 fw-bold d-flex"
        >
          <a
            class="text-decoration-none align-self-start"
            href="/appointments"
          >
            MY APPOINTMENTS
          </a>
        </li>
        <li
          class="list-group-item custom-link border-0 pl-2 fw-bold d-flex"
        >
          <a
            class="text-decoration-none align-self-start"
            href="/appointments/new"
          >
            MAKE APPOINTMENTS
          </a>
        </li>
        <li
          class="list-group-item custom-link border-0 pl-2 fw-bold d-flex"
        >
          <a
            class="text-decoration-none align-self-start"
            href="/doctors/new"
          >
            ADD DOCTOR
          </a>
        </li>
        <li
          class="list-group-item custom-link border-0 pl-2 fw-bold d-flex"
        >
          <a
            class="text-decoration-none align-self-start"
            href="/doctors/delete"
          >
            DELETE DOCTOR
          </a>
        </li>
        <li
          class="list-group-item custom-link border-0 pl-2 fw-bold d-flex"
        >
          <a
            class="text-decoration-none align-self-start"
            href="/doctors/delete"
          >
            LOG OUT
          </a>
        </li>
      </ul>
    </div>
    <div
      class="home_component"
    >
      <button
        aria-label="Slide left"
        class="flex"
        type="button"
      >
        <i
          class="fa fa-chevron-left"
          style="font-size: 2em; color: rgb(37, 200, 4);"
        />
      </button>
      <div
        class="slideshowSlider"
        id="slider"
      >
        <ul
          class="d-flex"
        />
      </div>
      <button
        aria-label="Slide Right"
        class="flex"
        type="button"
      >
        <i
          class="fa fa-chevron-right"
          style="font-size: 2em; color: rgb(37, 200, 4);"
        />
      </button>
    </div>
  </div>
</div>
`;
